pkg/exporter/syslogexporter/exporter_test.go

```diff
@@ -24,3 +76,92 @@ func TestInitExporter(t *testing.T) {
 		Format:   "rfc5424"}, createExporterCreateSettings())
 	assert.NoError(t, err)
 }
+
+func prepareExporterTest(t *testing.T, cfg *Config, userPort ...testPort) *exporterTest {
+	// Start a test syslog server
+	var port string
+	var addr net.TCPAddr
+	addr.IP = net.IP{127, 0, 0, 1}
+	addr.Port = 0
+	testServer, err := net.ListenTCP("tcp", &addr)
+	if err != nil {
+		t.Fatal("failed to start test syslog server:", err)
+	}
+	t.Cleanup(func() { testServer.Close() })
+	hostPort := testServer.Addr().String()
+	cfg.Endpoint, port, err = net.SplitHostPort(hostPort)
+	if err != nil {
+		t.Fatalf("could not parse port: %s", err)
+	}
+	if userPort[0].isIncorrect {
+		port = "112"
+	}
+	cfg.Port, err = strconv.Atoi(port)
+	if err != nil {
+		t.Fatalf("type error: %s", err)
+	}
+	exp, err := initExporter(cfg, createExporterCreateSettings())
+	require.NoError(t, err)
+	return &exporterTest{
+		srv: *testServer,
+		exp: exp,
+	}
+
+}
+
+func createTestConfig() *Config {
+	config := createDefaultConfig().(*Config)
+	config.Protocol = "tcp"
+	config.TLSSetting.Insecure = true
+	return config
+}
+
+func TestSyslogExportSuccess(t *testing.T) {
+	userPort := testPort{isIncorrect: false}
+	test := prepareExporterTest(t, createTestConfig(), userPort)
+	defer test.srv.Close()
+	go func() {
+		buffer := exampleLog()
+		logs := LogRecordsToLogs(buffer)
+		err := test.exp.pushLogsData(context.Background(), logs)
+		if err != nil {
+			fmt.Printf("could not send message: %s", err)
+		}
+	}()
+	err := test.srv.SetDeadline(time.Now().Add(time.Second * 1))
+	if err != nil {
+		t.Fatalf("could not accept connection: %s", err)
+	}
```

Comment:
Please use [require.NoError](https://pkg.go.dev/github.com/stretchr/testify/require#NoError) for error checking in tests
```suggestion
	require.NoError(t, err, "cannot set deadline")
```